docs/checklist_templates/eip_testing_checklist_template.md

```diff
@@ -0,0 +1,289 @@
+- [ ] Fork transition
+    - [ ] Verify that calling the precompile before its activation fork results in a valid call even for inputs that are expected to be invalid for the precompile.
+    - [ ] Verify that calling the precompile before its activation fork with zero gas results in a valid call.
+
+
+### Framework Changes
+
+- [ ] Add precompile address to relevant methods in the fork where the EIP is introduced in `src/ethereum_test_forks/forks/forks.py`
+
+## New System Contract
+
+### Test Vectors
+
+- [ ] Call contexts
+    - [ ] Normal call to system contract from contract
+    - [ ] Delegate call to system contract from contract
+    - [ ] Static call to system contract from contract
+    - [ ] Code call to system contract from contract
+    - [ ] System contract as transaction entry-point
+    - [ ] Call from initcode
+        - [ ] Contract creating transaction
+        - [ ] Contract creating opcode
+- [ ] Inputs
+    - [ ] Verify combinations of valid inputs to the system contract
+        - [ ] Verify interesting edge values given the system contract functionality.
+    - [ ] Verify all zeros input.
+    - [ ] Verify 2^N-1 where N is a single or multiple valid bit-lengths.
+- [ ] Out-of-bounds checks
+    - [ ] Verify if the system contract has out-of-bounds conditions in its inputs and verify:
+        - [ ] Max value for each input
+        - [ ] Max value + 1 for each input
+- [ ] Input Lengths
+    - [ ] Zero-length calldata.
+    - [ ] System contract has static-length input
+        - [ ] Correct static-length calldata
+        - [ ] Calldata too short, where the value represents a correct but truncated input to the contract.
+        - [ ] Calldata too long, where the value represents a correct input to the contract with padded zeros.
+    - [ ] System contract has dynamic-length input
+        - [ ] Verify correct System contract execution for valid lengths, given different inputs.
+        - [ ] Calldata too short, given different inputs, where the value represents a correct but truncated input to the contract.
+        - [ ] Calldata too long, given different inputs, where the value represents a correct input to the contract with padded zeros.
+- [ ] Excessive Gas Cases
+    - [ ] If possible, simulate a scenario where the execution of the contract at the end of the block execution by the system address would result in excessive gas usage (100 million gas or more).
+    - [ ] Verify spending all block gas in calls to system contract (100 million gas or more).
+- [ ] System Contract Deployment
+    - [ ] Verify block execution behavior after fork activation if the system contract has not been deployed.
+    - [ ] Verify deployment transaction results in the system contract being deployed to the expected address.
+- [ ] Contract Variations
+    - [ ] Verify execution of the different variations of the contract for different networks (if any) results in the expected behavior
```

Comment:
This is an interesting point, if we would assume that the bytecode on different networks is the same, the only opcode which comes to mind which might diverge chains is the `CHAINID` opcode at this point (but there might be more chain-specific opcodes, besides contextual opcodes like `TIMESTAMP`)

Reply:
Good point, I think this is something that needs to be brought up during audits of system contracts bytecode, CHAINID and similar codes should never be part of system contracts IMO.